com.unity.netcode.gameobjects/Tests/Runtime/NetworkBehaviourGenericTests.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Unity.Netcode.RuntimeTests
+{
+    /// <summary>
+    /// This class is for testing general fixes or functionality of NetworkBehaviours
+    /// </summary>
+    public class NetworkBehaviourGenericTests : BaseMultiInstanceTest
+    {
+        protected override int NbClients => 0;
+        public override IEnumerator Setup()
+        {
+            // Make sure we don't automatically start
+            m_BypassStartAndWaitForClients = true;
+
+            // Create the Host and add the SimpleNetworkBehaviour component
+            yield return base.Setup();
+        }
+
+        public class SimpleNetworkBehaviour : NetworkBehaviour
+        {
+        }
+
+        /// <summary>
+        /// This test validates a fix to NetworkBehaviour.NetworkObject when
+        /// the NetworkManager.LogLevel is set to Developer
+        /// Note: This test does not require any clients, but should not impact this
+        /// particular test if new tests are added to this class that do require clients
+        /// </summary>
+        [Test]
+        public void ValidateNoSpam()
```

Comment:
Do you really need the overhead of the multiinstancehelpers and actually starting a host in this test? Feels like you should be able to do this closer to a "unit" test... something like this:

```
        [Test]
        public void NetworkObjectGetterLogsMissingNetworkObject()
        {
            // Arrange NB with no NO, and Developer logging
            var gameObject = new GameObject();
            var nb = gameObject.AddComponent<EmptyNetworkBehaviour>();
            var nm = gameObject.AddComponent<NetworkManager>();
            nm.SetSingleton();
            nm.LogLevel = LogLevel.Developer;

            // Act
            var _ = nb.NetworkObject;

            // Assert - expected warning fired, tested generically to make the test less cumbersome
            LogAssert.Expect(LogType.Warning, new Regex(@"\[Netcode\].+"));
            LogAssert.NoUnexpectedReceived();
        }

        [Test]
        public void NetworkObjectGetterDoesNotLog()
        {
            // Arrange NB with an NO, and Developer logging
            var gameObject = new GameObject();
            var nb = gameObject.AddComponent<EmptyNetworkBehaviour>();
            gameObject.AddComponent<NetworkObject>();
            var nm = gameObject.AddComponent<NetworkManager>();
            nm.SetSingleton();
            nm.LogLevel = LogLevel.Developer;

            // Act
            var _ = nb.NetworkObject;
            
            LogAssert.NoUnexpectedReceived();
        }
```

or does that just cause problems "cleaning up" after the test?

Reply:
Yeah, I was thinking about making it a simple Unit test, but then realized we have no "generic" test for NetworkBehaviour where team members can add additional "generic tests and fix verifications".
I am more than happy to migrate this over to just a unit test (let me know your preference), but I would basically need to replicating the a bunch of the same code in the Setup and TearDown methods that comes with BaseMultiInstanceTest.

Reply:
It's up to you - I already approved the PR 👍